### ISSUE TITLE
[Bug] Windows accessibility issues

### ISSUE THREAD

**Issue by francipvb** [xamarin/Xamarin.Forms#7873]
### Description

I'm a screen reader user and I have some issues with windows universal implementation, particularly with binding-context backed controls such lists, but I found that table view renderer doesn't have a proper automation peer.

When I navigate to a table with NVDA, it says the full name of the type, for example "Xamarin.Forms.TextCell", "Xamarin.Forms.TableSection". Setting automation properties from xamarin.forms side doesn't change anything.

### Steps to Reproduce

Tested with Windows 10 Version 1903, but I expect to run in these issues in any version.

1. Create a new "Xamarin.Forms" blank project.
2. In the main page, add a table view with several demo sections and text cells.
3. Run the app from the UWP project and navigate the table with narrator or any screen reader.

### Expected Behavior

* Table section should render it's title as the automation name.
* Text cell should render it's text property as it's automation name and the details as an automation description.

### Actual Behavior

* Both render only automation name and it's value is just the full name of the type (xamarin.forms.TableSection, Xamarin.Forms.TextCell).

### Basic Information

- Version with issue: 4.2.0.848062
- Last known good version: None known
- IDE: Visual Studio 2019 (16.3)
- Platform Target Frameworks:
  - UWP: Minimum set to Windows 10 build 16299 and target to 1903, build 18362

**Comment by PureWeen** [comment]
https://github.com/dotnet/maui/issues/3998
